### DISCUSSION TITLE
Network error when running inference?

### DISCUSSION THREAD

**Discussion by junxiant** [Project-MONAI/MONAILabel#1449] [Q&A] [answered]
Hello,

I have been trying to run inference on CT scans with more slices (>200 slices), but each time i try to run the inference, i always get a network error, and an error parsing mask. This happens halfway through the inference.

![image](https://github.com/Project-MONAI/MONAILabel/assets/25401067/434c4cac-cd2e-4cfa-9221-5ad41f292bff)

When i checked the terminal, the inference still runs as normal.
![image](https://github.com/Project-MONAI/MONAILabel/assets/25401067/c7fee254-ec87-4fef-8bf6-2ebc5fd8b8dc)

And it even managed to complete the inference, with the results.
![image](https://github.com/Project-MONAI/MONAILabel/assets/25401067/48efddbb-8226-46b6-8c44-11bc38b00b54)

So when i checked out the console, turns out i was getting the "blocked by CORS policy".
![image](https://github.com/Project-MONAI/MONAILabel/assets/25401067/0d87fd77-73c9-4f42-9c09-0c4a62551657)

Has anyone faced this problem before? I'm using a custom domain for the xnat and the monailabel server.

The model works perfectly fine with other CT scans with fewer slices (eg. <70 slices)

Environment:
monai==1.1.0
monailabel==0.6.0

**Comment by SachidanandAlle** [comment]
Firstly you can upgrade monailabel version..
https://pypi.org/project/monailabel/#history

`pip install monailabel==0.7.0rc7`

Before we understand CORS problem.. from the logs, it's clear that server has done it's job and sent the result.
First you may want to check via (Try API http://monaiserver:port/ ) and run Inference API for your image, and check the inference result is all good..

I will leave it on you to identify why you are hitting with CORS issue (server running across different domain etc.. OHIF running in different box.. if-else conditions.. proxy, non-proxy etc..)

Until you fix the cross domain requests happening (CORS issue), you can run your browser with CORS disabled mode..
 for example for chrome: `chromium-browser --disable-web-security --user-data-dir=xyz`  (there are bunch of options available on google search for the different browser on this..

**Comment by junxiant** [comment]
Hmmm after updating to 0.7.0rc6, when I tried to connect to the monailabel server i got a 404 error.

![image](https://github.com/Project-MONAI/MONAILabel/assets/25401067/a44d975f-79ca-4cd6-8f76-a6383ceabd06)

Seems like its redirecting to //info which gives 404 error instead of /info?

It worked fine on 0.6.0, unless I'm supposed to change something after updating to 0.7.0rc6?

I'll try and fix the CORS from 0.6.0 for now.

**Comment by junxiant** [comment, marked as answer]
I have tested it out on the API and it seems like I am getting a 504 gateway timeout,
![image](https://github.com/Project-MONAI/MONAILabel/assets/25401067/e6274fd2-9fa8-49f9-91ca-fe2eb72b2231)

After checking out the load balancer it seems like aws was the issue. Changing the "Idle timeout" settings from 60 to 180 seems to have solved the issue. Looks like the load balancer was closing the connection to the server when the inference was running for more than 60 secs.
